ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,132 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      `);
+      await flush();
+      await layoutLoad(layout);
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+
+      layout.href = '';
+      await flush();
+
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.true;
+      expect(nuxeoInput.invalid).to.be.false;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.true;
+      expect(layout.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when native validation is invalid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+      expect(nuxeoInput.invalid).to.be.true;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.false;
+      expect(layout.validate.calledOnce).to.be.true;
+    });
+
+    test('Should not run custom validation when native validation fails', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout
+          href="${base}/layouts/dummy-layout.html"
+          model='{"text": "invalid", "data": "foo"}'
```

Comment:
But to achieve this (⬆️), I think we would this to look something like:
```suggestion
          model='{"text": "invalid"}'
```